docs/guide/interactions/views.rst

```diff
@@ -0,0 +1,467 @@
+
+Firstly we can create a component by subclassing the related component class. For a button this is the :class:`~discord.ui.Button` class.
+
+Our code might look like this:
+
+.. code-block:: python
+
+    class YesButton(discord.ui.Button[Confirm]):
+        def __init__(self) -> None:
+            super().__init__(
+                label='Yes',
+                style=discord.ButtonStyle.danger,
+            )
+
+        async def callback(self, interaction: discord.Interaction) -> None:
+            self.view.result = True
+            self.view.stop()
+            await interaction.response.defer()
+
+There's a bit to unpack here already, the ``__init__`` is setting the ``style`` and ``label`` parameters, which 
+set the colour and text of the button respectively.
+
+And then there's the :meth:`Button.callback() <discord.ui.Button.callback>` method, which is the method that will be called when the button is clicked.
+In this case it's being used to set the ``result`` attribute of the :class:`~discord.ui.View` subclass we created earlier.
+
+Now we've made our first component, we should add it to the view.
+
+The :class:`~discord.ui.View` class has a :meth:`~discord.ui.View.add_item` method, which takes a component as a parameter.
+so in our ``__init__`` method we can add the following after the call to the parent constructor.
+
+.. code-block:: python
+    :emphasize-lines: 7
+
+    class Confirm(discord.ui.View):
+
+        def __init__(self, user: discord.User, *, timeout: Optional[int] = None) -> None:
+            super().__init__(timeout=timeout)
+            self.user: discord.User = user
+            self.result: Optional[bool] = None
+            self.add_item(YesButton())
+
+        async def interaction_check(self, interaction: discord.Interaction) -> bool:
+            return interaction.user == self.user 
+
+
+Decorator-based Components
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In reality, our components are relatively simple, so we can use a helper function decorator instead.
+
+In this case, we're creating a button, so we can use the :func:`~discord.ui.button` decorator inside our ``View`` like so:
+
+.. code-block:: python
+    :emphasize-lines: 12-19
+
+    class Confirm(discord.ui.View):
+
+        def __init__(self, user: discord.User, *, timeout: Optional[int] = None) -> None:
+            super().__init__(timeout=timeout)
+            self.user: discord.User = user
+            self.result: Optional[bool] = None
+            self.add_item(YesButton())
+
+        async def interaction_check(self, interaction: discord.Interaction) -> bool:
+            return interaction.user == self.user 
+
+        @discord.ui.button(label='No', style=discord.ButtonStyle.primary)
+        async def no_button(
+            self,
+            interaction: discord.Interaction,
+            button: discord.ui.Button[Confirm],
+        ) -> None:
+            self.result = False
+            self.stop()
+            await interaction.response.defer()
```

Comment:
As noted, some sort of response needs to be sent to Discord. Most commonly, deleting the original response is what's implemented (or responding with a message saying that the user chose no)

Reply:
I added a note regarding this, could rework to cover more useful responses though.

Reply:
Sounds good